modules/space/shaders/fluxnodes_vs.glsl

```diff
@@ -0,0 +1,366 @@
+uniform int       colorMode;
+uniform sampler1D colorTable;
+uniform sampler1D colorTableCMR;
+uniform sampler1D colorTableEarth;
+uniform sampler1D colorTableFlow;
+uniform vec2      colorTableRange;
+
+// Uniforms needed for Particle Flow
+uniform vec4      flowColor;
+uniform int       particleSize;
+uniform int       particleSpeed;
+uniform int       particleSpacing;
+uniform bool      usingParticles;
+uniform bool      flowColoring;
+
+// Masking Uniforms
+uniform bool      usingMasking;
+uniform vec2      maskingRange;
+
+// Domain Uniforms
+uniform bool      usingDomain;
+uniform vec2      domainLimX;
+uniform vec2      domainLimY;
+uniform vec2      domainLimZ;
+uniform vec2      domainLimR;
+
+// Fluxnodes specific uniforms
+uniform float   nodeSize;
+uniform float   nodeSizeLargerFlux;
+uniform vec4    streamColor;
+uniform float   thresholdFlux;
+uniform float   filterLower;
+uniform float   filterUpper;
+uniform int     scalingMode;
+uniform int     nodeSkipMethod;
+uniform int     nodeSkip;
+uniform int     nodeSkipDefault;
+uniform int     nodeSkipEarth;
+uniform float   nodeSkipFluxThreshold;
+uniform float   nodeSkipRadiusThreshold;
+uniform float   fluxColorAlpha;
+uniform float   fluxColorAlphaIlluminance;
+uniform vec3    earthPos;
+uniform float   distanceThreshold;
+uniform int     enhanceMethod;
+uniform double  time;
+
+// Speicific uniforms for cameraperspective
+uniform float maxNodeDistanceSize;
+uniform float nodeDistanceThreshold;
+
+uniform vec3    cameraPos;
+//uniform vec2 screenSize;
+uniform bool    usingCameraPerspective;
+uniform bool    usingRadiusPerspective;
+uniform float   perspectiveDistanceFactor;
+
+uniform float   maxNodeSize;
+uniform float   minNodeSize;
+uniform bool    usingPulse;
+
+// Inputs
+// Should be provided in meters VaPosition
+layout(location = 0) in vec3 in_position;
+
+// The extra value used to color lines. Location must correspond to VaColor in 
+// renderablefluxnodes.h
+layout(location = 1) in float fluxValue;
+
+// The extra value used to mask out parts of lines. Location must correspond to 
+// VaFiltering in renderablefluxnodes.h
+layout(location = 2) in float rValue;
+
+// These should correspond to the enum 'ColorMode' in renderablefluxnodes.cpp
+const int colorByFluxValue  = 0;
+const int uniformColor     = 1;
+
+const int uniformskip = 0;
+const int fluxSkip = 1;
+const int radiusSkip = 2;
+
+const int fluxMode = 0;
+const int RFlux = 1;
+const int R2Flux = 2;
+const int log10RFlux = 3;
+const int lnRFlux = 4;
+
+const int sizeScaling = 0;
+const int colorTables = 1;
+const int sizeAndColor = 2;
+const int illuminance = 3;
+
+const float AUtoMeter = 149597871000.0;
+varying out vec4    vs_color;
+varying out float   vs_depth;
+varying out vec2    vs_st;
+varying out float   camera_IsCloseEnough;
+varying out float   vs_closeToEarth;
+varying out double  vs_time;
+varying out vec3    vs_camerapos;
+
+vec4 getTransferFunctionColor(sampler1D InColorTable) {
```

Comment:
Better not to define variable names starting with a capital letter.. only sadness and confusion ;)